v0/destinations/snapchat_custom_audience/transform.js

```diff
@@ -0,0 +1,173 @@
+const { validatePayload, validateFields } = require("./utils");
+
+/**
+ * Get access token to be bound to the event req headers
+ *
+ * Note:
+ * This method needs to be implemented particular to the destination
+ * As the schema that we'd get in `metadata.secret` can be different
+ * for different destinations
+ *
+ * @param {Object} metadata
+ * @returns
+ */
+const getAccessToken = metadata => {
+  // OAuth for this destination
+  const { secret } = metadata;
+  // we would need to verify if secret is present and also if the access token field is present in secret
+  if (!secret || !secret.access_token) {
+    throw new ErrorBuilder()
+      .setStatus(500)
+      .setMessage("Empty/Invalid access token")
+      .build();
+  }
+  return secret.access_token;
+};
+
+const generateResponse = (groupedData, schema, segmentId, metadata, type) => {
+  const payload = { users: [] };
+  const userPayload = { schema: [schema], data: groupedData };
+  payload.users.push(userPayload);
+  const response = defaultRequestConfig();
+  if (type === "remove") {
+    response.method = "DELETE";
+    payload.users[0].id = `${segmentId}`;
+  }
+
+  response.endpoint = `${BASE_URL}/segments/${segmentId}/users`;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  const accessToken = getAccessToken(metadata);
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json"
+  };
+
+  return response;
+};
+
+const getProperty = (schema, element) => {
+  let property;
+  switch (schema) {
+    case "email":
+      return element?.email;
+    case "phone":
+      property = element?.phone || element?.mobile;
+      return property;
+    case "mobileAdId":
+      property = element?.mobileId || element?.mobileAdId || element?.mobile_id;
+      return property;
+    default:
+      throw new CustomError("Invalid schema", 400);
+  }
+};
+
+const getHashedProperty = (schema, hashedProperty) => {
+  // Normalizing and hashing ref: https://marketingapi.snapchat.com/docs/#normalizing-hashing
+  switch (schema) {
+    case "email":
+      return sha256(hashedProperty.toLowerCase().trim());
+    case "phone":
+      return sha256(
+        hashedProperty
+          .toLowerCase()
+          .trim()
+          .replace(/^0+/, "")
+          .replace(/[^0-9]/g, "")
+      );
+    case "mobileAdId":
+      return sha256(hashedProperty.toLowerCase());
+    default:
+      throw new CustomError("Invalid schema", 400);
+  }
+};
+
+const getData = (userArray, schema, disableHashing) => {
+  const data = [[]];
+  let index = 0;
+  let hashedProperty;
+  userArray.forEach(element => {
+    hashedProperty = getProperty(schema, element);
+    if (!isDefinedAndNotNullAndNotEmpty(hashedProperty)) {
+      return;
+    }
+    if (!disableHashing) {
+      hashedProperty = getHashedProperty(schema, hashedProperty);
+    }
+    if (data[index].length >= 100000) {
+      data.push([]);
+      index += 1;
+    }
```

Comment:
Can you add some doc links regarding this?

Reply:
Added